globus_sdk/utils/token_storage.py

```diff
@@ -0,0 +1,217 @@
+import logging
+import time
+
+from globus_sdk.auth.client_types.native_client import NativeAppAuthClient
+from globus_sdk.config import get_parser
+from globus_sdk.exc import ConfigError, LoadedTokensExpired, RequestedScopesMismatch
+
+TOKEN_KEYS = [
+    "scope",
+    "access_token",
+    "refresh_token",
+    "token_type",
+    "expires_at_seconds",
+    "resource_server",
+]
+REQUIRED_KEYS = ["scope", "access_token", "expires_at_seconds", "resource_server"]
+CONFIG_TOKEN_GROUPS = "token_groups"
+
+logger = logging.getLogger(__name__)
+
+
+def save_tokens(tokens, config_section=None):
```

Comment:
`save_tokens` is intended to be used for arbitrary token storage, specifically by people registering their own scopes and wanting to easily save them for future use. This is most useful for people writing very simple clients or scripts. When somebody wants to write a custom client and save their own specific (non-token) related config-values, they should use their own config and not rely on the globus_sdk. A compromise there is to not expose this directly, and only allow it to be used indirectly by other SDK functionality.